File: Data Structures & Algorithms/remove-element/submission-3.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        wi = 0
        k = 0
        for ri in range(len(nums)):
            if nums[ri] != val:
                nums[wi] = nums[ri]
                wi += 1
            else:
                k += 1
        return len(nums) - k

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.rs
impl Solution {
  pub fn find_min(nums: Vec<i32>) -> i32 {
    let n = nums.len();
    let (mut l, mut r) = (0, n as i32 - 1);
    
    while l < r {
      let m = l + (r - l) / 2;
      if nums[m as usize] > nums[r as usize] {
        l = m + 1;
      } else {
        r = m;
      }
    }

    nums[r as usize]
  }
}